ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using ImperatorToCK3.CommonUtils;
+using commonItems;
+
+namespace ImperatorToCK3.CK3.Provinces {
+	public class ProvinceDetails {
+		// These values are open to ease management.
+		// This is a storage container for CK3::Province.
+		public string Culture { get; set; } = string.Empty;
+		public string Religion { get; set; } = string.Empty;
+		public string Holding { get; set; } = "none";
+		public List<string> Buildings { get; } = new();
+
+		public ProvinceDetails() { }
+		public ProvinceDetails(ProvinceDetails otherDetails) {
+			this.Culture = otherDetails.Culture;
+			Religion = otherDetails.Religion;
+			Holding = otherDetails.Holding;
+			Buildings = new List<string>(otherDetails.Buildings);
+		}
+		public ProvinceDetails(BufferedReader reader) {
+			var history = historyFactory.GetHistory(reader);
+			var date = new Date(867, 1, 1);
+
+			var cultureOpt = history.GetSimpleFieldValue("culture", date);
+			if (cultureOpt is not null) {
+				Culture = cultureOpt;
+			}
+			var religionOpt = history.GetSimpleFieldValue("religion", date);
+			if (religionOpt is not null) {
+				Religion = religionOpt;
+			}
+			var holdingOpt = history.GetSimpleFieldValue("holding", date);
+			if (holdingOpt is null) {
+				Logger.Warn("Province's holding can't be null!");
```

Comment:
Consider testing this

Reply:
I can't really test it, as the initial value for holding in historyFactory is "none", so the returned value will never be null. I added the null check to get rid of the IDE warning.